src/Kiota.Builder/CodeDOM/CodeClass.cs

```diff
@@ -202,7 +202,24 @@ public class ClassDeclaration : ProprietableBlockDeclaration
                 return currentParentClass.StartBlock.GetOriginalPropertyDefinedFromBaseType(serializationName);
         return default;
     }
+    public CodeProperty? FindPropertyByNameInTypeHierarchy(string propertyName)
```

Comment:
is there any specific reason why this method is defined in the start block instead of the code class?
Also moving it would allow to make it private.

Reply:
I still need to find my way in the code base. I followed the `GetOriginalPropertyDefinedFromBaseType` method. Maybe this method can be moved to `CodeClass` as well?

Reply:
yes, this should work. Can you give it a try please?

Reply:
Done